unit_tests/src/test_delay_hints_parser.cpp

```diff
@@ -262,36 +256,35 @@ TEST_CASE("#5503: getNextRepeatTime parses frequency correctly")
            {
                 .expected_action_code = 4,
                 .expected_next_time   = "10",
-                .expected_frequency   = double_until_success_or_until_time_string.replace(0, 2, "20")
+                .expected_frequency   = double_until_success_or_until_time_string_result
            }
         }},
         {"DOUBLE UNTIL SUCCESS OR <N> TIMES", {
             // Input
            {
                 .current_time = "0",
-                .frequency    = double_until_success_or_n_times
+                .frequency    = "10s DOUBLE UNTIL SUCCESS OR 20 TIMES"
            },
             // Output
            {
-                .expected_action_code = 0,
-                .expected_next_time   = "20",
-                .expected_frequency   = double_until_success_or_n_times
+                .expected_action_code = 4,
+                .expected_next_time   = "10",
+                .expected_frequency   = "20s DOUBLE UNTIL SUCCESS OR 19 TIMES. ORIGINAL TIMES=20"
```

Comment:
Is there a reason we do not have `double_until_success_or_n_times` and `double_until_success_or_n_times_result` strings here, similar to all the other cases?

Reply:
In this case, I didn't write a function to generate this type of directive string, so no intermediate variable was needed.
We should probably add a function to generate these types of strings too though.

I'm also thinking it might also be worth avoiding our use of `std::string_view` in our input and output structs too, just to avoid excess intermediate variables as a result of object lifetime considerations.

Reply:
Sure, that seems fine.